dotCMS/src/integration-test/java/com/dotcms/storage/StoragePersistenceAPITest.java

```diff
@@ -0,0 +1,279 @@
+     * Expected Result: We expect the default storage to be an instance of FileSystemStoragePersistenceAPIImpl
+     */
+    @Test
+    public void Test_Get_Default_Provider() {
+        final String stringProperty = Config.getStringProperty(DEFAULT_STORAGE_TYPE);
+        try {
+        //if there's a property already set we need to clean it so we can test that by default we will get FileSystem
+            Config.setProperty(DEFAULT_STORAGE_TYPE, null);
+            final StoragePersistenceProvider persistenceProvider = INSTANCE.get();
+            assertTrue(persistenceProvider
+                    .getStorage() instanceof FileSystemStoragePersistenceAPIImpl);
+        }finally {
+            Config.setProperty(DEFAULT_STORAGE_TYPE, stringProperty);
+        }
+    }
+
+    /**
+     * Given Scenario:
+     * Expected Result:
+     */
+    @Test
+    public void Test_Get_Provider_By_StorageType() {
+        final StoragePersistenceProvider persistenceProvider = INSTANCE.get();
+        assertTrue(persistenceProvider.getStorage(
+                StorageType.FILE_SYSTEM) instanceof FileSystemStoragePersistenceAPIImpl);
+        assertTrue(persistenceProvider
+                .getStorage(StorageType.DB) instanceof DataBaseStoragePersistenceAPIImpl);
+    }
+
+    /**
+     * Given scenario: For the same group we send two files of different type.
+     * Expected Result: The Group should be created then the file should be stored and then removed together with all of its elements.
+     * Same group is always send to test delete is effective.
+     */
+    @Test
+    @UseDataProvider("getMixedStorageTestCases")
+    public void Test_DBStorage_Push_File_Then_Recover_Then_Remove_Group(final TestCase testCase) {
+        final StoragePersistenceAPI storage = persistenceProvider.getStorage(testCase.storageType);
+        final String groupName = testCase.groupName;
+        assertFalse(storage.existsGroup(groupName));
+        assertTrue(storage.createGroup(groupName));
+        assertTrue(storage.listGroups().contains(groupName));
+        final String path = testCase.path;
+        final File pushFile = testCase.file;
+        final Object resultObject = storage.pushFile(groupName, path, pushFile, ImmutableMap.of());
+        assertNotNull(resultObject);
+        assertTrue(storage.existsGroup(groupName));
+        final File pullFile = storage.pullFile(groupName, path);
+        assertTrue(pullFile.exists());
+        assertTrue(pullFile.canRead());
+        assertTrue(pullFile.canWrite());
+        assertNotSame(pushFile, pullFile);
+        assertEquals(pullFile.length(), pushFile.length());
+        final int count = storage.deleteGroup(groupName);
+        assertEquals(1, count);
+        assertFalse(storage.existsGroup(groupName));
+        assertFalse(storage.existsObject(groupName, path));
+    }
+
+
+    @DataProvider
+    public static Object[] getMixedStorageTestCases() throws Exception {
```

Comment:
Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)